Microsoft.Research/Contracts/MsCorlib/System.Collections.ObjectModel.Collection.cs

```diff
@@ -0,0 +1,394 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Collections.ObjectModel
+{
+    // Summary:
+    //     Provides the base class for a generic collection.
+    //
+    // Type parameters:
+    //   T:
+    //     The type of elements in the collection.
+    public class Collection<T> : IList<T>, ICollection<T>, IEnumerable<T>, IList, ICollection, IEnumerable
```

Comment:
:bulb: It's difficult to tell while reading this contract class which methods are implementing interface members, and which methods are new for `Collection<T>`. For contracts, this distinction is important because only the latter may define contracts.